### PR TITLE
add scroll

### LINKED ISSUE #2430: sidebar: pages don't scroll
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Steps:

1. have more pages than the sidebar heigth can hold https://github.com/mui/mui-public/tree/master/tools-public
1. try to scroll

Screenshot 2023-08-04 at 14 55 58

### Current behavior 😯

It doesn't scroll

### Expected behavior 🤔

It should scroll

### Context 🔦

_No response_

### Your environment 🌎

_No response_

### PR DESCRIPTION
**Pull request by JerryWu1234**
fix #2430 
image

- [x] I've read and followed the [contributing guide](https://github.com/mui/mui-toolpad/blob/master/CONTRIBUTING.md#sending-a-pull-request) on how to create great pull requests.
- [x] I've updated the relevant documentation for any new or updated feature
- [x] I've linked relevant GitHub issue with "Closes #<issue id>"
- [x] I've added a visual demonstration under the form of a screenshot or video

### CODE REVIEW

File: packages/toolpad-app/src/toolpad/AppEditor/HierarchyExplorer/index.tsx [hunk 1]
```diff
@@ -227,7 +227,14 @@ export default function HierarchyExplorer({ className }: HierarchyExplorerProps)
   );
 
   return (
-    <HierarchyExplorerRoot data-testid="hierarchy-explorer" className={className}>
+    <HierarchyExplorerRoot
+      sx={{
+        height: '100%',
+        overflowY: 'auto',
```
Comment (Janpot) [inline packages/toolpad-app/src/toolpad/AppEditor/HierarchyExplorer/index.tsx:233]: Should this scroll horizontally as well?

Comment (JerryWu1234) [reply packages/toolpad-app/src/toolpad/AppEditor/HierarchyExplorer/index.tsx:233]: > Should this scroll horizontally as well?

So do you prefer `overflow: 'auto',`?

Comment (Janpot) [reply packages/toolpad-app/src/toolpad/AppEditor/HierarchyExplorer/index.tsx]: this tree can nest arbitrarily deep, so there may be edge cases where the indentation surpasses the width of the sidebar. horizontal overflow makes for good UX in those cases I think.

**Review by Janpot — APPROVED** [review]

**Review by bharatkashyap — APPROVED** [review]
👌

**Comment by oliviertassinari** [comment]
@JerryWu1234 Thanks!

**Pull request merged**